core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala

```diff
@@ -898,18 +1083,17 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext with B
     }
 
     // Here is the main check of this test -- we have the same offers again, and we schedule it
-    // successfully.  Because the scheduler first tries to schedule with locality in mind, at first
-    // it won't schedule anything on executor1.  But despite that, we don't abort the job.  Then the
-    // scheduler tries for ANY locality, and successfully schedules tasks on executor1.
+    // successfully.  Because the scheduler tries to schedule with locality in mind, at first
+    // it won't schedule anything on executor1.  But despite that, we don't abort the job.
     val secondTaskAttempts = taskScheduler.resourceOffers(offers).flatten
-    assert(secondTaskAttempts.size == 2)
-    secondTaskAttempts.foreach { taskAttempt => assert("executor1" === taskAttempt.executorId) }
```

Comment:
the change in behavior here is because it hasn't met the locality delay yet to go to executor1 now - and before it used to? please explain.

Reply:
will follow up on this as I update tests.

Reply:
originally after the first resourceOffer, the locality level is set to ANY due there not being any remaining NODE_LOCAL tasks (set to ANY by `TaskSetManager.getAllowedLocalityLevel`)

with new behavior, the locality level is set to NODE_LOCAL since it is overridden in `TaskSchedulerImpl.resourceOffers`.
Normally the behavior would be the same because a subsequent call to `TaskSetManager.getAllowedLocalityLevel` would increase the locality level, but due to the tasks failing, there still remain NODE_LOCAL tasks by the time it is called again.

Reply:
I see we are still on node local locality since we didn't reject any so we have to wait the timeout here so these are rejected at this point.